<a href="https://colab.research.google.com/github/Va-un/IMDB_movie_analysis/blob/main/IMDB_movie_analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Data analysis of 1000 IMDB movies (2006-2016) and finding the trends of what people like and genre they prefer to watch

Dataset  contains all the data of IMDB from the year 2006 to 2016. The era where some of the greatest movies were made

In [133]:
#Downloading all the required documentation.
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import plotly.express as px
import plotly.graph_objects as go

# Data cleaing and processing

In [134]:
df = pd.read_csv("/content/IMDB-Movie-Data.csv")
df

Rank                    Title                     Genre  \
0       1  Guardians of the Galaxy   Action,Adventure,Sci-Fi   
1       2               Prometheus  Adventure,Mystery,Sci-Fi   
2       3                    Split           Horror,Thriller   
3       4                     Sing   Animation,Comedy,Family   
4       5            Suicide Squad  Action,Adventure,Fantasy   
..    ...                      ...                       ...   
995   996     Secret in Their Eyes       Crime,Drama,Mystery   
996   997          Hostel: Part II                    Horror   
997   998   Step Up 2: The Streets       Drama,Music,Romance   
998   999             Search Party          Adventure,Comedy   
999  1000               Nine Lives     Comedy,Family,Fantasy   

                                           Description              Director  \
0    A group of intergalactic criminals are forced ...            James Gunn   
1    Following clues to the origin of mankind, a te...          Ridley Scott   
2    Three girls are kidnapped by a man with a diag...    M. Night Shyamalan   
3    In a city of humanoid animals, a hustling thea...  Christophe Lourdelet   
4    A secret government agency recruits some of th...            David Ayer   
..                                                 ...                   ...   
995  A tight-knit team of rising investigators, alo...             Billy Ray   
996  Three American college students studying abroa...              Eli Roth   
997  Romantic sparks occur between two dance studen...            Jon M. Chu   
998  A pair of friends embark on a mission to reuni...        Scot Armstrong   
999  A stuffy businessman finds himself trapped ins...      Barry Sonnenfeld   

                                                Actors  Year  \
0    Chris Pratt, Vin Diesel, Bradley Cooper, Zoe S...  2014   
1    Noomi Rapace, Logan Marshall-Green, Michael Fa...  2012   
2    James McAvoy, Anya Taylor-Joy, Haley Lu Richar...  2016   
3    Matthew McConaughey,Reese Witherspoon, Seth Ma...  2016   
4    Will Smith, Jared Leto, Margot Robbie, Viola D...  2016   
..                                                 ...   ...   
995  Chiwetel Ejiofor, Nicole Kidman, Julia Roberts...  2015   
996  Lauren German, Heather Matarazzo, Bijou Philli...  2007   
997  Robert Hoffman, Briana Evigan, Cassie Ventura,...  2008   
998  Adam Pally, T.J. Miller, Thomas Middleditch,Sh...  2014   
999  Kevin Spacey, Jennifer Garner, Robbie Amell,Ch...  2016   

     Runtime (Minutes)  Rating   Votes  Revenue (Millions)  Metascore  
0                  121     8.1  757074              333.13       76.0  
1                  124     7.0  485820              126.46       65.0  
2                  117     7.3  157606              138.12       62.0  
3                  108     7.2   60545              270.32       59.0  
4                  123     6.2  393727              325.02       40.0  
..                 ...     ...     ...                 ...        ...  
995                111     6.2   27585                 NaN       45.0  
996                 94     5.5   73152               17.54       46.0  
997                 98     6.2   70699               58.01       50.0  
998                 93     5.6    4881                 NaN       22.0  
999                 87     5.3   12435               19.64       11.0  

[1000 rows x 12 columns]

In [135]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 12 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Rank                1000 non-null   int64  
 1   Title               1000 non-null   object 
 2   Genre               1000 non-null   object 
 3   Description         1000 non-null   object 
 4   Director            1000 non-null   object 
 5   Actors              1000 non-null   object 
 6   Year                1000 non-null   int64  
 7   Runtime (Minutes)   1000 non-null   int64  
 8   Rating              1000 non-null   float64
 9   Votes               1000 non-null   int64  
 10  Revenue (Millions)  872 non-null    float64
 11  Metascore           936 non-null    float64
dtypes: float64(3), int64(4), object(5)
memory usage: 93.9+ KB


We can say that data set contains about 12 columns and we can see that **there are some null values** as well in Revenue and Metascore

In [136]:
df.rename(columns={'Runtime (Minutes)':'Runtime' ,
                   'Revenue (Millions)': 'Revenue'},inplace=True, errors='raise')

In [137]:
df.isnull().sum()

Rank             0
Title            0
Genre            0
Description      0
Director         0
Actors           0
Year             0
Runtime          0
Rating           0
Votes            0
Revenue        128
Metascore       64
dtype: int64

Here we cans see that revenue has aboout 128 null and Metascore has about 64 null.

In [138]:
df.Revenue.fillna(value = 0,inplace = True )
df.Metascore.fillna(value = 0,inplace = True )


In [139]:
df.isna().sum()

Rank           0
Title          0
Genre          0
Description    0
Director       0
Actors         0
Year           0
Runtime        0
Rating         0
Votes          0
Revenue        0
Metascore      0
dtype: int64

Here we replace all the NAN values with 0 since  revenue and Metascore is thye row specific and each  movie  has its own specific value for those and these values have no relation to the nearby values

In [140]:
df['Genre'] = df.Genre.str.split(',')
df['Actors'] = df.Actors.str.split(',')


Converting the string into array so that thry can be easily accessible for using array funcion for splitting them for visualisation

Deleting the non required column so that we can work on it more easily

In [141]:
df.describe()

Rank         Year      Runtime       Rating         Votes  \
count  1000.000000  1000.000000  1000.000000  1000.000000  1.000000e+03   
mean    500.500000  2012.783000   113.172000     6.723200  1.698083e+05   
std     288.819436     3.205962    18.810908     0.945429  1.887626e+05   
min       1.000000  2006.000000    66.000000     1.900000  6.100000e+01   
25%     250.750000  2010.000000   100.000000     6.200000  3.630900e+04   
50%     500.500000  2014.000000   111.000000     6.800000  1.107990e+05   
75%     750.250000  2016.000000   123.000000     7.400000  2.399098e+05   
max    1000.000000  2016.000000   191.000000     9.000000  1.791916e+06   

           Revenue    Metascore  
count  1000.000000  1000.000000  
mean     72.337960    55.210000  
std     100.320314    22.030598  
min       0.000000     0.000000  
25%       3.352500    43.000000  
50%      37.145000    58.000000  
75%      99.177500    71.000000  
max     936.630000   100.000000

looking at above data we can say


1.   rank list is till 1000
2.   year  goes from 2006 to 2016
3.   Runtime is minimum for an hour and  goes till 3 hours  with average of hour.
4.   Rating is from 1-9
5. votes are from 61 to 1791916 with mean 169808.255
6. Revenue goes at average of 72.3 million.
7. Metascore of mean 55








In [142]:
del df['Description']

# Data analysis

In [143]:
fig = px.box(df, x="Year", y="Revenue",title='Revenue of top movies by year')
fig.show()

From above graph we can see average income per year and we can draw the following interferance in following points-
NOTE- since most of the values where NAN converted to 0 there might be some changes.
1. Maximum income was done in year 2015 of 936.6 million followed by movie in 2009 with revenue of 760 million
2. most of income values fall between about 80 -100
3. most movies in 2016 has made 3.1 million


In [144]:
dfg = df.explode("Genre")

In [145]:
fig = px.histogram(dfg, x="Genre",color = "Year",title='Movies in each  genre')
fig.show()

From above we can say that drama is the most liked genre followed by action  and then comedy. Drama has 500 plus,Action has 300 and comedy about 270.
Musical is the least liked with about only 5 likes


In [146]:
dfg = df.explode("Genre")
fig = px.bar(dfg, x="Genre",y = "Revenue",color = "Year",title='Total revenue earned by top 1000 movies by each genre')
fig.show()


Highest Revenue is earned by adventure movies followed by action and then drama.Adventure has earned  about 42k with action  about 37k.
Musical was produced least so has earned the least

In [147]:
fig = px.histogram(dfg, x="Director",color = "Year",title='Directors with thier movie releases')
fig.show()

We can say that most top  movies are released by Ridney Scott.

In [148]:
pop = dfg[dfg.Director == "Ridley Scott"].Genre.value_counts()
cop = ["Drama","Adventure","Action","Sci-Fi","Crime","Romance","Mystery","Biography","Thriller","Comedy"]
fig = px.pie(df, values=pop, names=cop, title='Ridley Scott most produced genre')
fig.show()

Ridley Scott mostly produces  Dramas followed by adventure.

In [149]:
dfa =df.explode('Actors')

Now we are done with genre let let us move to  actors analysis

In [150]:
fig = px.histogram(dfa, x="Actors",color = "Year")
fig.show()

We can see that Mark Wahlberg & Christian Bale  has made the most top movies so we can assume that he  is a favourate actor

In [151]:

a = dfg.groupby(['Director']).agg({'Revenue':'mean','Metascore':'mean'})
x = a.sort_values(by=['Revenue','Metascore'], ascending = False).head(10)
x.reset_index(level=0, inplace=True)

fig = go.Figure()
fig.add_trace(go.Bar(
    x=x.Director,
    y=x.Revenue,
    name='Revenue',
    marker_color='blue'
))
fig.add_trace(go.Bar(
    x=x.Director,
    y=x.Metascore,
    name='Metascore',
    marker_color='green'
))
fig.update_layout(barmode='group', xaxis_tickangle=-45,title='Revenue and metascore of the Directors')
fig.show()

We can say that James Cameron has the higest revenue and has made the most liked movies,followed by Colin Trevorrow	